### PR TITLE
feat: add a/b test for customer logos on signup page

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Problem

[Add a new a/b test](https://us.posthog.com/project/2/experiments/24705) building upon the current [generic-signup-benefits](https://us.posthog.com/project/2/experiments/2671) with:

(new one updated to only run on desktop/tablet)

1. control - normal language around product analytics

Screenshot 2024-03-21 at 11 52 56 AM

2. generic language test - more generic language for all products

Screenshot 2024-03-21 at 11 53 04 AM

3. logos test - custom logos instead of just names

Screenshot 2024-03-21 at 11 52 46 AM

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## TODOs
 - [ ] turn on new a/b test
 - [ ] delete old a/b test

## Does this work well for both Cloud and self-hosted?

it doesn't have an impact

### CODE REVIEW

File: frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png
Comment (xrdt) [inline frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png:1]: Can we confirm the logos look good on dark mode, as well? If you're going to add the logos, I suppose we should have storybook snapshots for each variation of the feature flag.

File: frontend/src/scenes/authentication/signup/SignupContainer.tsx [hunk 1]
```diff
@@ -50,51 +56,82 @@ export function SignupContainer(): JSX.Element | null {
...
+                {
+                    benefit: 'Join industry leaders that run on PostHog',
+                    description:
+                        'Airbus, Hasura, Y Combinator, Staples, and thousands more trust PostHog as their Product OS.',
+                },
+            ]
+        case 'logos':
+            return [
+                {
+                    benefit: 'Free for 1M events every month',
```
Comment (xrdt) [inline frontend/src/scenes/authentication/signup/SignupContainer.tsx:86]: Is `1M events free every month` more clear?

File: frontend/src/scenes/authentication/signup/SignupContainer.tsx [hunk 2]
```diff
@@ -50,51 +56,82 @@ export function SignupContainer(): JSX.Element | null {
...
+    switch (signupBenefitsFlag) {
+        case 'generic-language':
+            return [
+                {
+                    benefit: 'Free usage every month - even on paid plans',
+                    description: '1M free events, 5K free session recordings, and more. Every month. Forever.',
+                },
+                {
+                    benefit: 'Start collecting data immediately',
+                    description: 'Integrate with developer-friendly APIs or low-code web snippet.',
```
Comment (xrdt) [inline frontend/src/scenes/authentication/signup/SignupContainer.tsx:75]:
```suggestion
                    description: 'Integrate with developer-friendly APIs or a low-code web snippet.',
```

**Review by xrdt — APPROVED** [review]

File: frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png
Comment (zlwaterfield) [reply frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png:1]: Yes, will do!

File: frontend/src/scenes/authentication/signup/SignupContainer.tsx (hunk 1 above)
Comment (zlwaterfield) [reply frontend/src/scenes/authentication/signup/SignupContainer.tsx:86]: Yep, I like that more 👍

Comment (zlwaterfield) [inline frontend/src/scenes/authentication/signup/SignupContainer.tsx:86]:
```suggestion
                    benefit: '1M events free every month',
```

File: frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png
Comment (zlwaterfield) [reply frontend/__snapshots__/scenes-other-signup--self-hosted-sso--light.png:1]: Turns out auto pages are not dark mode friendly. See if statement here: https://github.com/PostHog/posthog/blob/master/frontend/src/layout/navigation-3000/themeLogic.ts#L28-L34. I'll look into that more but we can ignore for now.

**Pull request merged**